### PR TITLE
arceos: add raw backtrace qemu test cases

### PR DESCRIPTION
**Pull request by Jiaxin2006**
Tracking issue: #146

## What
- Add ArceOS QEMU raw backtrace test cases covering:
  - normal unwind (expects BT 0/BT 1)
  - bad frame pointer handling (expects BT_ERROR invalid_fp)
  - basic raw report smoke
- Add a manual-only panic-path switch for `arceos-backtrace-raw-basic` (feature `panic-path`) and a dedicated build config `build-x86_64-unknown-none-panic.toml` for local verification.

## Why
- Strengthen E2E coverage for raw backtrace block format and edge conditions.
- Keep panic-path verification out of CI (runner default fail pattern), but still make it easy to run locally.

## How to verify
- Run QEMU tests:
  - `cargo xtask arceos test qemu --arch x86_64 --test-group rust --test-case backtrace-raw-normal`
  - `cargo xtask arceos test qemu --arch x86_64 --test-group rust --test-case backtrace-raw-badfp`
  - `cargo xtask arceos test qemu --arch x86_64 --test-group rust --test-case backtrace-raw-basic`

## Notes
- Host-side symbolization is handled in #635 and is intentionally not included in this PR.

### CODE REVIEW

File: test-suit/arceos/rust/backtrace-raw-normal/build-x86_64-unknown-none.toml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+features = ["ax-std"]
+log = "Warn"
+max_cpu_num = 1
+
+[env]
```
Comment (ZR233) [inline test-suit/arceos/rust/backtrace-raw-normal/build-x86_64-unknown-none.toml:5]: 此 `RUSTFLAGS` 设置会完全覆盖 ArceOS 构建系统生成的 RUSTFLAGS（包含链接器脚本路径等关键标志），导致链接阶段出现 `relocation R_X86_64_64 cannot be used against local symbol` 错误。

本地验证：
```
cargo xtask arceos test qemu --arch x86_64 --test-group rust --test-case backtrace-raw-normal
→ error: could not compile `arceos-backtrace-raw-normal`
```

建议移除 `[env] RUSTFLAGS` 覆盖，改用构建系统的 `DWARF=y` 配置来启用 frame pointers。如果 xtask 不支持通过 build config 设置 DWARF，需要在 xtask 层面添加支持。

File: test-suit/arceos/rust/backtrace-raw-badfp/build-x86_64-unknown-none.toml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+features = ["ax-std"]
+log = "Warn"
+max_cpu_num = 1
+
+[env]
```
Comment (ZR233) [inline test-suit/arceos/rust/backtrace-raw-badfp/build-x86_64-unknown-none.toml:5]: 同上，`RUSTFLAGS` 覆盖导致构建失败：
```
cargo xtask arceos test qemu --arch x86_64 --test-group rust --test-case backtrace-raw-badfp
→ error: could not compile `arceos-backtrace-raw-badfp`
```

File: test-suit/arceos/rust/backtrace-raw-basic/qemu-x86_64.toml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    "128M",
+    "-smp",
+    "4",
+    "-nographic",
+    "-serial",
+    "mon:stdio",
+]
+uefi = false
+to_bin = false
+success_regex = [
```
Comment (ZR233) [inline test-suit/arceos/rust/backtrace-raw-basic/qemu-x86_64.toml:16]: 此 success_regex 要求匹配 `^BT\s+0\s+ip=0x[0-9a-fA-F]+\s+fp=0x[0-9a-fA-F]+`，但本地运行时实际输出为：
```
BACKTRACE_BEGIN kind=raw arch=x86_64 alloc=false dwarf=false
BT_ERROR invalid_fp
BACKTRACE_END
```

没有 `BT 0` 行，该 pattern 不应匹配。测试虽然通过了，但 success_regex 并未真正验证到 backtrace 帧的正确性。需要确认 xtask runner 是否要求所有 success_regex pattern 全部匹配，以及这个测试在无 frame pointers 时的预期行为。

File: test-suit/arceos/rust/backtrace-raw-basic/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+app! {
+
+#[cfg(feature = "ax-std")]
+extern crate ax_std as std;
+
+use axbacktrace::Backtrace;
+use std::println;
+
+fn c() {
```
Comment (ZR233) [inline test-suit/arceos/rust/backtrace-raw-basic/src/main.rs:32]: 这个测试名为 `backtrace-panic` 但实际上不触发 panic，而是直接调用 `Backtrace::report("raw")`。同时 `lang_items.rs` 中 panic 处理器已改为使用 `Backtrace::report("panic")`，但该 panic 路径没有任何 QEMU 测试覆盖。

建议：
1. 如果要测试 panic 路径的 backtrace 报告，应保留触发 panic 的行为并调整 fail_regex
2. 如果只测试 raw report 格式，建议重命名为 `backtrace-raw-basic` 之类的名称
3. 考虑补充一个专门测试 panic handler 路径 backtrace 输出的用例

**Review by ZR233 — CHANGES_REQUESTED** [review]
## 审查总结

这个 PR 为 axbacktrace 添加了机器可读的原始 backtrace 报告格式（`BacktraceReport`），并新增了三个 QEMU 测试用例来验证报告格式和 FP 边界条件。

实现逻辑整体合理：`BacktraceReport` 输出结构化的 `BACKTRACE_BEGIN/END` 格式，包含 FP 链校验（对齐、单调性、跳跃距离），适合 host 端符号化工具解析。`read_current_fp()` 从内联汇编重构为独立函数也符合代码复用原则。

但存在以下阻塞问题需要修复：

### 阻塞问题

1. **`backtrace-raw-normal` 和 `backtrace-raw-badfp` 构建失败**

   两个新增测试用例的 `build-*.toml` 中 `[env] RUSTFLAGS = "-C force-frame-pointers=yes -C debuginfo=2 -C strip=none"` 会完全覆盖 ArceOS 构建系统的 RUSTFLAGS（包括链接器脚本路径等），导致链接错误：
   ```
   rust-lld: error: relocation R_X86_64_64 cannot be used against local symbol; recompile with -fPIC
   ```

   本地验证命令及结果：
   ```
   cargo xtask arceos test qemu --arch x86_64 --test-group rust --test-case backtrace-raw-normal
   # → error: could not compile `arceos-backtrace-raw-normal`
   ```

2. **`backtrace-panic` 的 success_regex 与实际输出不匹配**

   `qemu-*.toml` 中 `success_regex` 要求匹配 `^BT\s+0\s+ip=0x...` 但实际输出是 `BT_ERROR invalid_fp`，没有 `BT 0` 行。本地运行该测试时输出为：
   ```
   BACKTRACE_BEGIN kind=raw arch=x86_64 alloc=false dwarf=false
   BT_ERROR invalid_fp
   BACKTRACE_END
   ```

   虽然测试通过了（可能是 runner 只检查部分 pattern），但 success_regex 声称要求至少一帧，实际上并未验证到。

3. **`backtrace-panic` 测试名不反映实际行为**

   该测试在 commit 3 中最初会触发 panic 来测试 backtrace 报告，但在 commit 5 中被改为直接调用 `Backtrace::report("raw")`，不再走 panic 路径。同时 `fail_regex = ["(?i)\\bpanic(?:ked)?\\b"]` 会在出现 panic 关键字时判定失败。

   然而 `lang_items.rs` 中的 panic 处理器已被改为使用 `Backtrace::report("panic")`。这意味着关键的 panic 路径 backtrace 报告没有任何 QEMU 测试覆盖。

### 非阻塞建议

- 考虑将 `backtrace-panic` 重命名为更准确的名称（如 `backtrace-raw-basic`），或恢复其原始的 panic 路径测试意图。
- 如果需要多架构覆盖 raw report 格式，当前的 `backtrace-panic` 可以保留多架构配置，但建议名称更清晰。

### 验证结果

| 命令 | 结果 |
|------|------|
| `cargo fmt --check` | ✅ 通过 |
| `cargo clippy -p axbacktrace --all-features` | ✅ 通过 |
| `cargo test -p axbacktrace` | ✅ 5/5 通过 |
| `cargo xtask arceos test qemu --arch x86_64 -c backtrace-panic` | ✅ 通过（但 success_regex 不完全匹配） |
| `cargo xtask arceos test qemu --arch x86_64 -c backtrace-raw-normal` | ❌ 构建失败 |
| `cargo xtask arceos test qemu --arch x86_64 -c backtrace-raw-badfp` | ❌ 构建失败 |

CI 状态：pending（仅有 Detect changed paths 在运行，其余 skipped）。

**Comment by Jiaxin2006** [comment]
@ZR233 感谢 review！你提到的阻塞点已在最新提交中处理：

- backtrace-raw-normal / backtrace-raw-badfp：已移除 build-*.toml 里 `[env] RUSTFLAGS=...` 的覆盖，改为 `DWARF="y"` 由构建系统注入必要 rustflags（含 frame pointers / debuginfo / strip 配置），避免覆盖链接脚本相关 flags。
- success_regex 假绿：已将相关 QEMU 用例的 `success_regex` 合并为单条多行正则，避免多条 success_regex 的 OR 行为导致“未真正验证到 BT 0/预期块仍通过”。
- backtrace-panic 名不副实：已将该用例重命名为 backtrace-raw-basic，明确职责为 raw report 的 smoke（不再暗示 panic-path 覆盖）。

本地已用 `cargo xtask arceos test qemu --arch x86_64 --test-group rust --test-case ...` 验证 backtrace-raw-basic / backtrace-raw-normal / backtrace-raw-badfp 通过。

**Review by ZR233 — APPROVED** [review]
